dbms/src/Dictionaries/HTTPDictionarySource.cpp

```diff
@@ -31,22 +34,65 @@ HTTPDictionarySource::HTTPDictionarySource(const DictionaryStructure & dict_stru
 
 HTTPDictionarySource::HTTPDictionarySource(const HTTPDictionarySource & other)
     : log(&Logger::get("HTTPDictionarySource")),
+    update_time{other.update_time},
     dict_struct{other.dict_struct},
     url{other.url},
+    update_field{other.update_field},
+    date{other.date},
     format{other.format},
     sample_block{other.sample_block},
     context(other.context),
     timeouts(ConnectionTimeouts::getHTTPTimeouts(context.getSettingsRef()))
 {
 }
 
+void HTTPDictionarySource::setDate()
+{
+    if (!hasUpdateField())
+        return;
+    else if ((hasUpdateField() && date == "0000-00-00%2000:00:00"))
+    {
+        auto tmp_time = update_time;
+        update_time = std::chrono::system_clock::now();
+        auto duration = std::chrono::duration_cast<std::chrono::seconds>(update_time - tmp_time);
+        time_t hr_time = std::chrono::system_clock::to_time_t(update_time) - duration.count() - 1;
+        char buffer [80];
+        struct tm * timeinfo;
+        timeinfo = localtime (&hr_time);
+        strftime(buffer, 80, "%Y-%m-%d%%20%H:%M:%S", timeinfo);
+        std::string str_time(buffer);
+        date = str_time;
+    }
+    else
+    {
+        auto tmp_time = update_time;
+        update_time = std::chrono::system_clock::now();
+        auto duration = std::chrono::duration_cast<std::chrono::seconds>(update_time - tmp_time);
+        time_t hr_time = std::chrono::system_clock::to_time_t(update_time) - duration.count() - 1;
+        char buffer [80];
+        struct tm * timeinfo;
+        timeinfo = localtime (&hr_time);
+        strftime(buffer, 80, "%Y-%m-%d%%20%H:%M:%S", timeinfo);
+        std::string str_time(buffer);
+        date = str_time;
+        url_update = url + update_field + date;
```

Comment:
It's also strange. Probably, we want to add `?update_field=date` or something similar.